src/Network/Algorand/Definitions.hs

```diff
@@ -0,0 +1,105 @@
+-- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+-- | Definitions to work with Algo, grouped in this module for convenience.
+module Network.Algorand.Definitions
+  ( Network (..)
+  , NetworkError (..)
+  , networkParser
+
+  , Host
+  , DefaultHost (..)
+  , getDefaultHost
+  ) where
+
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+
+import Control.Exception.Safe (Exception)
+import Data.Aeson (FromJSON (..), ToJSON (..))
+import Data.Functor ((<&>))
+import Data.Map.Strict (Map)
+import Data.Text (Text)
+
+-- | Supported networks.
+data Network
+  = MainnetV1
+  | TestnetV1
+  | BetanetV1
+  deriving (Eq, Ord)
+
+-- | Mapping from `Network` to it's name
+networkToName :: Network -> Text
+networkToName = \case
+  MainnetV1 -> "mainnet-v1.0"
+  TestnetV1 -> "testnet-v1.0"
+  BetanetV1 -> "betanet-v1.0"
+
+networkParser :: String -> Either String Network
+networkParser = \case
+  "mainnet-v1.0" -> Right MainnetV1
+  "testnet-v1.0" -> Right TestnetV1
+  "betanet-v1.0" -> Right BetanetV1
+  x -> Left $ "Unmapped network name: " <> x
+
+instance Show Network where
+  show = T.unpack . networkToName
+
+instance ToJSON Network where
+  toJSON = toJSON . networkToName
+
+instance FromJSON Network where
+  parseJSON o = do
+    value <- parseJSON o <&> networkParser
+    case value of
+      Right v -> pure v
+      Left e -> error e
```

Comment:
`error` is ok for programmer mistakes (places that shouldn't be reachable regardless of user input). But when you parse JSON, in general case you can't know that it's valid. So I think `error` is not appropriate here. You can use `fail` for example.